### PR TITLE
Latest and minimum dependency test

### PR DESCRIPTION
**Pull request by praveenkuttappan**
This PR is to add new tox environment to test a package against latest released dependent package on PyPI or minimum required version of a package on PyPI

Changes:
 - Two new tox environment
- script to install either latest or minimum required version of dependent azure sdk packages

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -116,7 +116,7 @@ jobs:
         CoverageArg: $(CoverageArg)
         PythonVersion: $(PythonVersion)
         BuildTargetingString: ${{ parameters.BuildTargetingString }}
-        ToxTestEnv: 'whl,sdist'
+        ToxTestEnv: 'whl,sdist,latestdependency,minimumdependency'
```
Comment (scbedd) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml]: Leave this for testing, I don't want to introduce additional build time _yet_.

File: eng/tox/install_depend_packages.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+sys.path.append(setup_parser_path)
+from setup_parser import get_install_requires
+
+DEV_REQ_FILE = "dev_requirements.txt"
+NEW_DEV_REQ_FILE = "new_dev_requirements.txt"
+PKGS_TXT_FILE = "packages.txt"
+EXCLUDED_PKGS = ['azure-common',]
+
+def install_dependent_packages(setup_py_file_path, dependency_type, temp_dir):
+    # This method identifies latest/ minimal version of dependent packages and install them from pyPI
```
Comment (scbedd) [inline eng/tox/install_depend_packages.py]: nit: `install them from pyPI` -> `installs them from pyPI`

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml (hunk 1 above)
Comment (praveenkuttappan) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml]: I will remove this additional test. Added it to verify on pipelines. Re -running entire repo test on my machine now. I will remove these tests and push once test is completed.

File: eng/tox/tox.ini [hunk 3]
```diff
@@ -168,4 +168,37 @@ commands =
...
       {[testenv]default_pytest_params} \
       {posargs} \
-      {toxinidir}
\ No newline at end of file
+      {toxinidir}
+
+
+[deptestcommands]
+commands =
+    {envbindir}/python {toxinidir}/../../../eng/tox/install_depend_packages.py -t {toxinidir} -d {env:DEPENDENCY_TYPE:} -w {envtmpdir}
```
Comment (scbedd) [inline eng/tox/tox.ini]: So glad this panned out. Excellent!

File: eng/tox/install_depend_packages.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+setup_parser_path = os.path.abspath(
+    os.path.join(os.path.abspath(__file__), "..", "..", "versioning")
+)
+sys.path.append(setup_parser_path)
+from setup_parser import get_install_requires
+
+DEV_REQ_FILE = "dev_requirements.txt"
+NEW_DEV_REQ_FILE = "new_dev_requirements.txt"
+PKGS_TXT_FILE = "packages.txt"
+EXCLUDED_PKGS = ['azure-common',]
```
Comment (scbedd) [inline eng/tox/install_depend_packages.py]: Why do we exclude this here, but still build the package? Probably just forgetting a situation you already introduced.

File: eng/tox/tox.ini (hunk 3 above)
Comment (scbedd) [reply eng/tox/tox.ini]: Also noticing that you're using `envtmpdir` that's the way to be!
